prometheus/src/test/scala/filodb/prometheus/parse/ParserSpec.scala

```diff
@@ -414,6 +414,30 @@ class ParserSpec extends AnyFunSpec with Matchers {
     parseSuccessfully(bob.append("\"}").toString())
   }
 
+  it("parse subqueries") {
+    parseSubquery("min_over_time( rate(http_requests_total[5m])[30m:1m] )")
+    parseSubquery("max_over_time( deriv( rate(distance_covered_meters_total[1m])[5m:1m] )[10m:] )")
+    parseSubquery("max_over_time((time() - max(foo) < 1000)[5m:10s] offset 5m)")
+    parseSubquery("avg_over_time(rate(demo_cpu_usage_seconds_total[1m])[2m:10s])")
+
+    parseSubquery("foo[5m:1m]")
+    parseSubquery("foo[5m:]")
+    parseSubquery("max_over_time(rate(foo[5m])[5m:1m])")
+    parseSubquery("max_over_time(sum(foo)[5m:1m])")
+    parseSubquery("sum(foo)[5m:1m]")
+    parseSubquery("log2(foo)[5m:1m]")
+    parseSubquery("log2(foo)[5m:]")
+    parseSubquery("(foo + bar)[5m:1m]")
+    parseSubquery("sum_over_time((foo + bar)[5m:1m])")
+    parseSubquery("avg_over_time(max_over_time(rate(foo[5m])[5m:1m])[10m:2m])")
+
+    parseSubqueryError("log2(foo)[5m][5m:1m]")
+    parseSubqueryError("sum(foo)[5m]")
+    // FIXME: these should be uncommented when subquery support is finished
```

Comment:
I added it and the test passes (the parse fails as expected). Is this correct behavior or is a subquery of a subquery allowed? I assume not since the output of a subquery is a range query, but the input is an instant query.

Reply:
Something is still a bit odd with the test result. This needs more investigation.

Reply:
Okay, I've fixed ParserSpec such that this query test fails, which is expected for now. That is, the query parses, but it's otherwise illegal. Making the parser accurately track data types is tricky and is harder to maintain.